File: ts/2391_minimum-amount-of-time-to-collect-garbage.ts
function garbageCollection(garbage: string[], travel: number[]): number {
    let sumM = 0, sumP = 0, sumG = 0
    let pathM = [0, 0], pathP = [0, 0], pathG = [0, 0]
    garbage.forEach((g, idx) => {
        for (let c of g) {
            switch (c) {
                case 'M':
                    pathM[1] = idx
                    sumM++
                    break;
                case 'P':
                    pathP[1] = idx
                    sumP++
                    break;
                case 'G':
                    pathG[1] = idx
                    sumG++
                    break;
            }
        }
    })
    const s = [0]
    for (let t of travel) {
        s.push(s[s.length - 1] + t)
    }
    let pathSum = 0
    for (let [from, to] of [pathM, pathP, pathG]) {
        pathSum += s[to] - s[from]
    }
    return sumM + sumP + sumG + pathSum
};

console.log(garbageCollection(["G","P","GP","GG"], [2,4,3]))
console.log(garbageCollection(["MMM","PGM","GP"], [3,10]))

File: csharp/2391_minimum-amount-of-time-to-collect-garbage.cs
namespace L2391;

public class Solution
{
    public int GarbageCollection(string[] garbage, int[] travel)
    {
        int sumM = 0, sumP = 0, sumG = 0;
        (int from, int to) pathM = (from: 0, to: 0), pathP = (from: 0, to: 0), pathG = (from: 0, to: 0);
        for (int i = 0; i < garbage.Length; i++)
        {
            foreach (var c in garbage[i])
            {
                switch (c)
                {
                    case 'M':
                        sumM++;
                        pathM.to = i;
                        break;
                    case 'P':
                        sumP++;
                        pathP.to = i;
                        break;
                    case 'G':
                        sumG++;
                        pathG.to = i;
                        break;
                }
            }
        }
        var s = new int[garbage.Length];
        s[0] = 0;
        for (int i = 1; i < garbage.Length; i++) {
            s[i] = s[i - 1] + travel[i - 1];
        }
        return s[pathM.to] - s[pathM.from] + s[pathP.to] - s[pathP.from] + s[pathG.to] - s[pathG.from] + sumM + sumP + sumG;
    }
}